lib/alerts/informed_entity.ex

```diff
@@ -72,21 +70,24 @@ defmodule Alerts.InformedEntity do
   Otherwise the nil can match any value in the other InformedEntity.
 
   """
-  @spec match?(IE.t(), IE.t()) :: boolean
-  def match?(%IE{} = first, %IE{} = second) do
+  @spec match?(Alerts.InformedEntity.t(), Alerts.InformedEntity.t()) :: boolean
+  def match?(%__MODULE__{} = first, %__MODULE__{} = second) do
     share_a_key?(first, second) && do_match?(first, second)
   end
 
+  @spec mapsets_match?(MapSet.t(), MapSet.t()) :: boolean()
   def mapsets_match?(%MapSet{} = a, %MapSet{} = b)
-      when a == @empty_activities or b == @empty_activities,
+      when a == %MapSet{} or b == %MapSet{},
       do: true
 
   def mapsets_match?(%MapSet{} = a, %MapSet{} = b), do: has_intersect?(a, b)
 
   defp has_intersect?(a, b), do: Enum.any?(a, &(&1 in b))
 
   defp do_match?(f, s) do
-    @fields
+    __MODULE__.__struct__()
+    |> Map.keys()
+    |> List.delete(:__struct__)
```

Comment:
📓 Instead of duplicating the list of fields and hard-coding them, I get them from the struct.